packages/grafana-ui/src/components/Table/types.ts

```diff
@@ -99,6 +99,8 @@ export interface Props {
   enableSharedCrosshair?: boolean;
   // The index of the field value that the table will initialize scrolled to
   initialRowIndex?: number;
+  // Map of react elements to render within column headers, index is field id
+  columnHeaders?: Map<number, React.ReactElement>;
```

Comment:
why use different pattern for passing this than we do for cellComponent? 

https://github.com/grafana/grafana/blob/cfcf03bf7af4f1928ed728aa38eb86c874ba46cc/packages/grafana-ui/src/components/Table/Table.story.tsx#L286